source-code/core2/src/config/schema.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from "zod"
+
+const PluginSchema = z.object({
+	module: z.string(),
+	options: z.record(z.string()),
+})
+
+export const InlangConfigSchema = z.object({
+	sourceLanguageTag: z.string(),
+	languageTags: z.array(z.string()),
+	plugins: z.array(PluginSchema),
```

Comment:
## Use key based plugin config schema or array?  

Current type: 

```
{
  plugins: [
    { 
      module: "https://..........."
      options: {

      }
    }
  ]
}
```

### Problem (?): 

Removing plugins manually is a bit cumbersome because one needs to identify the module link. 

### Question 

We could make manual removal of plugins easier. BUT, the plugin key (id) can (will) go out of sync with the returned plugin id of a plugin. 

```
{
  plugins: {
    inlang.plugin-i18next: {
      module: "https://......",
      options: {

      }
    }
  }
}

```

I vote for not worth increasing manual removal of plugins at the risk of out of sync plugin ids.

Reply:
We already changed the naming/ids of plugins pretty often. That will happen in the future, too.
I think links are the better choice. 👍

Reply:
> We already changed the naming/ids of plugins pretty often

- I do not think that this should be a common practice (how often do npm packages names change?)
- module link & identifier ("inlang.plugin-i18next") are even duplicating themselves
- giving an identifier & a dedicated field for version (for example in options) is also a way to go that we could differentiate behavior with versions in the future

Reply:
> I do not think that this should be a common practice anymore

Although it maybe is not a problem for us, in an ecosystem with hundreds of community plugins, this can become an issue.

Reply:
> Although it maybe is not a problem for us, in an ecosystem with hundreds of community plugins, this can become an issue.

In addition, transfering plugins would become unfeasible. 

Seems like we agree with "array of objects" with no identifiers because plugins define the ids. 

Vote with 👍 or 👎

Reply:
I can get myself into links, but just wanted to add that with npm, there is also no way to rename an already published package. They resolve it like: https://stackoverflow.com/a/38632439 

with `options.version` & `identifier`, we would follow npm registry behavior until we have our own registry.

Reply:
> with options.version & identifier, we would follow npm registry behavior until we have our own registry. 

NPM is too complex. Resolve from a link is vastly simpler. Resolving because we agreed.